File: app/schemas/filing.py
from uuid import UUID

from pydantic import BaseModel


class MaterialValidation(BaseModel):
    material_id: UUID
    name: str
    is_qualified: bool
    has_signature: bool
    issues: list[str]


class FilingPackResult(BaseModel):
    filing_doc_id: UUID
    materials_count: int
    qualified_count: int
    missing_signatures: list[str]
    ready: bool

File: frontend/src/types/filing.ts
export interface MaterialValidation {
  material_id: string;
  name: string;
  is_qualified: boolean;
  has_signature: boolean;
  issues: string[];
}

export interface FilingPackResult {
  filing_doc_id: string;
  materials_count: number;
  qualified_count: number;
  missing_signatures: string[];
  ready: boolean;
}
